app/scripts/metamask-controller.js

```diff
@@ -4665,6 +4680,42 @@ export default class MetamaskController extends EventEmitter {
           },
         );
       }
+
+      ///: BEGIN:ONLY_INCLUDE_IF(multi-srp)
+      const client = new KeyringInternalSnapClient({
+        messenger: this.controllerMessenger,
+        snapId: SOLANA_WALLET_SNAP_ID,
+      });
+      // TODO: Use `withKeyring` instead.
+      const keyring = await this.getSnapKeyring();
+      for (let index = 0; ; index++) {
+        const discovered = await client.discoverAccounts(
+          [SolScope.Mainnet, SolScope.Testnet, SolScope.Devnet],
+          entropySource,
+          index,
+        );
+
+        // We stop discovering accounts if none got discovered for that index.
+        if (discovered.length === 0) {
+          break;
+        }
+
+        await Promise.allSettled(
+          discovered.map(async (discoveredAccount) => {
+            const options = {
+              derivationPath: discoveredAccount.derivationPath,
+              entropySource,
+            };
+
+            await keyring.createAccount(SOLANA_WALLET_SNAP_ID, options, {
+              displayConfirmation: false,
+              displayAccountNameSuggestion: false,
+              setSelectedAccount: false,
+            });
+          }),
+        );
+      }
+      ///: END:ONLY_INCLUDE_IF
```

Comment:
What do you think about extracting this into a different method to avoid bloating this one and the file?